projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -1741,25 +1747,34 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy {
      * @memberof IgxColumnComponent
      */
     public unpin(index?: number): boolean {
-        if (this.grid) {
-            this.grid.endEdit(true);
-        }
+        const grid = (this.grid as any);
         if (!this._pinned) {
             return false;
         }
 
         if (this.parent && this.parent.pinned) {
             return this.topLevelParent.unpin(index);
         }
-
-        const grid = (this.grid as any);
         const hasIndex = index !== undefined;
         if (hasIndex && (index < 0 || index >= grid._unpinnedColumns.length)) {
             return false;
         }
 
+        // correct unpinnedIndex, if it exceeds unpinned columns length
+        if (this._unpinnedIndex > grid._unpinnedColumns.length) {
+            this._unpinnedIndex = grid._unpinnedColumns.length;
+        }
+
         index = (index !== undefined ? index :
             this._unpinnedIndex !== undefined ? this._unpinnedIndex : this.index);
+
+        const args: IPinColumnCancellableEventArgs = { column: this, insertAtIndex: index, isPinned: false, cancel: false };
+        this.grid.onColumnPinning.emit(args);
+
+        if (args.cancel) { return; }
```

Comment:
we should change isPinned to false if the event is canceld in order to reciev the right state of the column